src/main/java/net/minecraftforge/fml/client/gui/screen/DefaultConfigScreen.java

```diff
@@ -0,0 +1,250 @@
+
+package net.minecraftforge.fml.client.gui.screen;
+
+import com.electronwill.nightconfig.core.UnmodifiableConfig;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.widget.TextFieldWidget;
+import net.minecraft.client.gui.widget.Widget;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+import net.minecraftforge.common.ForgeConfigSpec;
+import net.minecraftforge.fml.client.gui.IPressHandler;
+import net.minecraftforge.fml.client.gui.widget.ExtendedButton;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Map;
+
+/**
+ * Default screen for configs from the mod menu. Can handle boolean, integer, long, double and string properties.
+ */
+@OnlyIn(Dist.CLIENT)
+public class DefaultConfigScreen extends Screen {
+    protected Screen previousScreen;
+    protected ForgeConfigSpec configSpec;
+    protected static final Logger LOGGER = LogManager.getLogger();
+
+    /**
+     * @param titleIn will be displayed on top
+     */
+    public DefaultConfigScreen(ITextComponent titleIn, Screen previousScreen, ForgeConfigSpec forgeConfigSpec) {
+        super(titleIn);
+        this.previousScreen = previousScreen;
+        this.configSpec = forgeConfigSpec;
+    }
+
+    @Override
+    protected void init() {
+        super.init();
+        Map<String, Object> specMap = configSpec.getSpec().valueMap();
+        Map<String, Object> valueMap = configSpec.getValues().valueMap();
+        int spacing = 30;
+        int x = 6;
+        for (String key : specMap.keySet()) {
+            Object value = specMap.get(key);
+            //nested configs
+            if (value instanceof UnmodifiableConfig) {
+                UnmodifiableConfig unmodifiableConfig = (UnmodifiableConfig) value;
+                Map<String, Object> entries = unmodifiableConfig.valueMap();
+                UnmodifiableConfig actualValues = (UnmodifiableConfig) valueMap.get(key);
+                for (String k : entries.keySet()) {
+                    Object v = entries.get(k);
+                    spacing = buildElements(spacing, x, actualValues, k, v);
+                }
+            } else {
+                //immediate config
+                spacing = buildElements(spacing, x, configSpec.getValues(), key, value);
+            }
+        }
+    }
+
+    /**
+     * Builds screen elements from the config
+     *
+     * @return vertical offset for the next screen element
+     */
+    @SuppressWarnings("unchecked")
+    protected int buildElements(int spacing, int x, UnmodifiableConfig actualValues, String k, Object v) {
+        if (v instanceof ForgeConfigSpec.ValueSpec) {
+            ForgeConfigSpec.ValueSpec valueSpec = (ForgeConfigSpec.ValueSpec) v;
+            Object specValue = actualValues.get(k);
+            Object valueRange = valueSpec.getRange();
```

Comment:
Why on earth are you typing this as `Object`? Adjust the visibility on `Range` if you need it.